tests/backend/test_postgres_db.py

```diff
@@ -30,39 +60,92 @@ async def test_postgresql_connection_not_ok(
-    postgresql_url, backend_factory, caplog, unused_tcp_port, autojump_clock
-):
-    tries_number = 4
-    tries_sleep = 3
-    postgresql_url = f"postgresql://localhost:{unused_tcp_port}/dummy"
-    with pytest.raises(OSError) as exc:
-        with trio.fail_after((tries_number - 1) * tries_sleep + 1):
-            async with backend_factory(
-                config={
-                    "db_url": postgresql_url,
-                    "db_first_tries_number": tries_number,
-                    "db_first_tries_sleep": tries_sleep,
-                }
-            ):
-                pass
-    if platform.system() == "Darwin":
-        errno = 61
-    else:
-        errno = 111
-    assert f"[Errno {errno}] Connect call failed" in str(exc.value)
-    records = records_filter_debug(caplog.records)
-    assert len(records) == 4
-    for record in records[:3]:
-        assert record.levelname == "WARNING"
-        assert "initial db connection failed" in record.message
-    assert records[3].levelname == "ERROR"
-    assert "initial db connection failed" in records[3].message
+async def test_retry_policy_no_retry(postgresql_url, unused_tcp_port, asyncio_loop):
+    host = "localhost"
+    port = unused_tcp_port
+    app_config = BackendConfig(
+        administration_token="s3cr3t",
+        db_min_connections=1,
+        db_max_connections=5,
+        debug=False,
+        blockstore_config=PostgreSQLBlockStoreConfig(),
+        email_config=None,
+        backend_addr=None,
+        forward_proto_enforce_https=None,
+        ssl_context=False,
+        spontaneous_organization_bootstrap=False,
+        organization_bootstrap_webhook_url=None,
+        db_url=postgresql_url,
+    )
+
+    # No retry
+    retry_policy = RetryPolicy(maximum_attempts=0, pause_before_retry=0)
+
+    # Expect a connection error
+    with pytest.raises(ConnectionError):
+        async with trio.open_nursery() as nursery:
+            # Run backend in the background
+            nursery.start_soon(
+                lambda: _run_backend(
+                    host, port, ssl_context=False, retry_policy=retry_policy, app_config=app_config
+                )
+            )
+            # Connect to PostgreSQL database
+            async with triopg.connect(postgresql_url) as conn:
+                # Wait for the backend to be connected
+                pid, = await wait_for_listener(conn)
+                # Terminate the backend listener connection
+                value, = await conn.fetchrow(f"SELECT pg_terminate_backend($1)", pid)
+                assert value
+                # The _run_backend should raise a Connection error and cancel the nursery
+                await trio.sleep(3)
+                assert False
+
+
+@pytest.mark.trio
+@pytest.mark.postgresql
+async def test_retry_policy_allow_retry(postgresql_url, unused_tcp_port, asyncio_loop):
+    host = "localhost"
+    port = unused_tcp_port
+    app_config = BackendConfig(
```

Comment:
Shouldn't we create a `backend_config_factory` (or a `backend_config_default_params`) fixture for this ? (would be shared between this file's tests and `backend_factory`)